src/clock.rs

```diff
@@ -125,12 +129,12 @@ impl ClockSetup {
         self
     }
 
-    pub fn freeze(self) -> Clocks {
+    pub fn freeze(self) -> Result<Clocks> {
         let coreclk = self.coreclk.unwrap_or(HFXCLK);
         let pclk = self.pclk.unwrap_or(HFXCLK / 2);
 
-        let core_pll = PllConfig::calculate(HFXCLK, coreclk).unwrap();
-        let hfpclk_pll = PllConfig::calculate(HFXCLK, pclk * 2).unwrap();
+        let core_pll = PllConfig::calculate(HFXCLK, coreclk)?;
+        let hfpclk_pll = PllConfig::calculate(HFXCLK, pclk * 2)?;
```

Comment:
Error propagation here loses context: you can't tell anymore which of the two methods failed. Error type returned from `freeze` is not helpful in this situation.

Reply:
Good catch; each should add context for the calculation being performed